tests/unit/src/main/scala/tests/BaseDapSuite.scala

```diff
@@ -8,8 +8,48 @@ import scala.concurrent.Future
 import scala.meta.internal.metals.debug.DebugWorkspaceLayout
 import scala.meta.internal.metals.debug.Stoppage
 import scala.meta.internal.metals.debug.TestDebugger
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.GlobalTrace
+import scala.meta.internal.metals.debug.DebugProtocol
+import scala.util.Failure
+import scala.util.Success
 
 abstract class BaseDapSuite(suiteName: String) extends BaseLspSuite(suiteName) {
+
+  private val dapClient =
+    GlobalTrace.protocolTracePath(DebugProtocol.clientName)
+  private val dapServer =
+    GlobalTrace.protocolTracePath(DebugProtocol.serverName)
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    dapClient.touch()
+    dapServer.touch()
+  }
+
+  protected def logDapTraces(): Unit = {
+    scribe.warn("The DAP test failed, printing the traces")
+    scribe.warn(dapClient.toString() + ":")
+    scribe.info(dapClient.readText)
+    scribe.warn(dapServer.toString() + ":")
+    scribe.info(dapServer.readText)
+  }
+
+  override def munitTestTransforms: List[TestTransform] =
+    super.munitTestTransforms :+
+      new TestTransform("Print DAP traces", { test =>
+        test.withBody(() =>
+          test
+            .body()
+            .andThen {
+              case Failure(exception) =>
+                logDapTraces()
+                exception
+              case Success(value) => value
+            }(munitExecutionContext)
```

Comment:
@olafurpg @gabro IT seems that if we use something different than `munitExecutionContext` we get:
```
X tests.debug.StackFrameDapSuite.afterAll  0.002s java.lang.IllegalArgumentException: requirement failed
2020-04-15T12:00:59.8858923Z [error]     at scala.Predef$.require(Predef.scala:268)
2020-04-15T12:00:59.8859382Z [error]     at scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:55)
2020-04-15T12:00:59.8859785Z [error]     at scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:875)
2020-04-15T12:00:59.8860176Z [error]     at scala.concurrent.BatchingExecutor$Batch.blockOn(BatchingExecutor.scala:96)
2020-04-15T12:00:59.8872621Z [error]     at scala.concurrent.package$.blocking(package.scala:146)
2020-04-15T12:00:59.8873105Z [error]     at scala.concurrent.java8.FuturesConvertersImpl$CF.get(FutureConvertersImpl.scala:92)
2020-04-15T12:00:59.8873645Z [error]     at scala.meta.internal.metals.MetalsLanguageServer.cancel(MetalsLanguageServer.scala:89)
2020-04-15T12:00:59.8874081Z [error]     at scala.meta.internal.metals.MetalsLanguageServer.cancelAll(MetalsLanguageServer.scala:97)
2020-04-15T12:00:59.8874432Z [error]     at tests.BaseLspSuite.afterAll(BaseLspSuite.scala:46)
2020-04-15T12:00:59.8874769Z [error]     at munit.MUnitRunner.$anonfun$runAfterAll$3(MUnitRunner.scala:160)
```
any idea how we can make it clearer? Maybe we can have an additional mapping that doesn't take executionContext?

Reply:
Created an issue https://github.com/scalameta/munit/issues/127